matchers/matcher.go

```diff
@@ -0,0 +1,218 @@
+package matchers
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/erpc/erpc/common"
+)
+
+// ConfigMatcher handles matching logic for requests and responses using MatcherConfig
+type ConfigMatcher struct {
+	configs []*common.MatcherConfig
+}
+
+// NewConfigMatcher creates a new matcher with the given configs
+func NewConfigMatcher(configs []*common.MatcherConfig) *ConfigMatcher {
+	return &ConfigMatcher{
+		configs: configs,
+	}
+}
+
+// MatchRequest evaluates if a request matches based on the configs
+func (m *ConfigMatcher) MatchRequest(networkId, method string, params []interface{}, finality common.DataFinalityState) common.MatchResult {
+	if len(m.configs) == 0 {
+		return common.MatchResult{Matched: true, Action: common.MatcherInclude}
+	}
+
+	for i := len(m.configs) - 1; i >= 0; i-- {
+		config := m.configs[i]
+		if m.matchConfig(config, networkId, method, params, finality, false) {
+			return common.MatchResult{Matched: true, Action: config.Action}
+		}
+	}
+
+	// Default behavior if no matchers match - this should not happen if catch-all rules are properly configured
+	// We determine the default based on the intended behavior:
+	// - If the first matcher (processed last) is an include, default to exclude
+	// - If the first matcher (processed last) is an exclude, default to include
+	// - Otherwise, default to exclude for safety (fail-closed)
+	if len(m.configs) > 0 {
+		firstMatcher := m.configs[0]
+		if firstMatcher.Action == common.MatcherInclude {
+			return common.MatchResult{Matched: false, Action: common.MatcherExclude}
+		} else if firstMatcher.Action == common.MatcherExclude {
+			return common.MatchResult{Matched: false, Action: common.MatcherInclude}
+		}
+	}
+
+	// Fallback to exclude for safety
+	return common.MatchResult{Matched: false, Action: common.MatcherExclude}
+}
+
+// MatchForCache evaluates if a response should be cached
+func (m *ConfigMatcher) MatchForCache(networkId, method string, params []interface{}, finality common.DataFinalityState, isEmptyish bool) common.MatchResult {
```

Comment:
Any reason for this? 

We can have `matchers.Match(cfg, req, resp)` exposed from the package, very simple and works in every place